python/paddle/distribution/binomial.py

```diff
@@ -0,0 +1,276 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Sequence
+
+import paddle
+from paddle.distribution import distribution
+
+
+class Binomial(distribution.Distribution):
+    r"""
+    The Binomial distribution with size `total_count` and `probability` parameters.
+
+    In probability theory and statistics, the binomial distribution is the most basic discrete probability distribution defined on :math:`[0, n] \cap \mathbb{N}`,
+    which can be viewed as the number of times a potentially unfair coin is tossed to get heads, and the result
+    of its random variable can be viewed as the sum of a series of independent Bernoulli experiments.
+
+    The probability mass function (pmf) is
+
+    .. math::
+
+        pmf(x; n, p) = \frac{n!}{x!(n-x)!}p^{x}(1-p)^{n-x}
+
+    In the above equation:
+
+    * :math:`total_count = n`: is the size, meaning the total number of Bernoulli experiments.
+    * :math:`probability = p`: is the probability of the event happening in one Bernoulli experiments.
+
+    Args:
+        total_count(int|Tensor): The size of Binomial distribution which should be greater than 0, meaning the number of independent bernoulli
+            trials with probability parameter :math:`p`. The data type will be converted to 1-D Tensor with paddle global default dtype if the input
+            :attr:`probability` is not Tensor, otherwise will be converted to the same as :attr:`probability`.
+        probability(float|Tensor): The probability of Binomial distribution which should reside in [0, 1], meaning the probability of success
+            for each individual bernoulli trial. If the input data type is float, it will be converted to a 1-D Tensor with paddle global default dtype.
+
+    Examples:
+        .. code-block:: python
+
+            >>> import paddle
+            >>> from paddle.distribution import Binomial
+            >>> paddle.set_device('cpu')
+            >>> paddle.seed(100)
+            >>> rv = Binomial(100, paddle.to_tensor([0.3, 0.6, 0.9]))
+
+            >>> print(rv.sample([2]))
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[31., 62., 93.],
+             [29., 54., 91.]])
+
+            >>> print(rv.mean)
+            Tensor(shape=[3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [30.00000191, 60.00000381, 90.        ])
+
+            >>> print(rv.entropy())
+            Tensor(shape=[3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [2.94053698, 3.00781751, 2.51124287])
+    """
+
+    def __init__(self, total_count, probability):
```

Comment:
use `probs` instead of `probability` will be better? consistent with other similar APIs like ` paddle.distribution.Bernoulli`,  ` paddle.distribution.Geometric` and `paddle.distribution.Multinomial`?

Reply:
[rfc](https://github.com/PaddlePaddle/community/blob/master/rfcs/APIs/20230926_api_design_for_binomial.md?plain=1#L484) use `probs` which is inconsistent with the code

Reply:
已修改